File: NumberTheory/seiveOfEratosthenes/prime.js
(function() {
    (function main() {

        let n = 10;
        let is_prime = [];
        for (let i=0; i<=n; i++) {
            is_prime.push(true);
        }
        is_prime[0] = false;
        is_prime[1] = false;
        
        for (let i = 2; i<=n; i++) {
            if (is_prime[i] && i*i <= n) {
                for (let j = i*i; j<=n; j+=i) {
                    is_prime[j] = false;
                }
            }
        }

        let cnt = 0;
        for (let i = 2; i<=n; i++) {
            if (is_prime[i]) cnt++;
        }

        console.log(cnt);

    }());
}());

File: NumberTheory/seiveOfEratosthenes/prime.cpp
#include<iostream>
#include<vector>
using std::cin;
using std::cout;
using std::endl;
using std::vector;

int main() {
    int n;
    cin >> n;

    vector<bool> is_prime(n + 1, true);
    is_prime[0] = is_prime[1] = false;
    for (int i{2}; i * i <= n; i++) {
        if (is_prime[i] && (long long) i*i <= n) {
            for (int j{i * i}; j <= n; j += i) is_prime[j] = false;
        }
    }

    int cnt {};
    for (int i{2}; i<=n; i++) {
        if (is_prime[i]) cnt++;
    }
    cout << cnt << endl;
}
